FireRender.Maya.Src/FireRenderExportCmd.cpp

```diff
@@ -416,171 +418,205 @@ MStatus FireRenderExportCmd::doIt(const MArgList & args)
 			}
 
-			// Refresh the context so it matches the
-			// current animation state and start the render.
-			northStarContextPtr->Freshen();
+			MString namePattern;
+			argData.getFlagArgument(kPadding, 0, namePattern);
+			unsigned int framePadding = 0;
+			argData.getFlagArgument(kPadding, 1, framePadding);
+
+			// create rprs context
+			frw::RPRSContext rprsContext;
 
-			// update file path
-			std::wstring newFilePath; 
-			if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+			// process each frame
+			for (int frame = firstFrame; frame <= lastFrame; ++frame)
 			{
-				//GetPattern();
-				std::wstring name_regex;
-				std::wstring frame_regex(L"#");
-				std::wstring extension_regex;
-				GetUINameFrameExtPattern(name_regex, extension_regex);
-				std::wstring pattern = namePattern.asWChar();
+				// Move the animation to the next frame.
+				if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+				{
+					MTime time;
+					time.setValue(static_cast<double>(frame));
+					MStatus isTimeSet = MGlobal::viewFrame(time);
+					CHECK_MSTATUS(isTimeSet);
+				}
 
-				// Replace extension at first, because it shouldn't match name_regex or frame_regex for given .rpr format
-				std::wstring result = std::regex_replace(pattern, std::wregex(extension_regex), fileExtension);
+				// Refresh the context so it matches the
+				// current animation state and start the render.
+				northStarContextPtr->Freshen();
 
-				std::wstringstream frameStream;
-				frameStream << std::setfill(L'0') << std::setw(framePadding) << frame;
-				result = std::regex_replace(result, std::wregex(frame_regex), frameStream.str().c_str());
+				// update file path
+				std::wstring newFilePath;
+				if (isSequenceExportEnabled && !isAnimationAsSingleFileEnabled)
+				{
+					//GetPattern();
+					std::wstring name_regex;
+					std::wstring frame_regex(L"#");
+					std::wstring extension_regex;
+					GetUINameFrameExtPattern(name_regex, extension_regex);
+					std::wstring pattern = namePattern.asWChar();
 
-				// Replace name after all operations, because it could match frame or extension regex
-				result = std::regex_replace(result, std::wregex(name_regex), fileName);
+					// Replace extension at first, because it shouldn't match name_regex or frame_regex for given .rpr format
+					std::wstring result = std::regex_replace(pattern, std::wregex(extension_regex), fileExtension);
 
-				newFilePath = result.c_str();
-			}
-			else
-			{
-				newFilePath = fileName + L"." + fileExtension;
+					std::wstringstream frameStream;
+					frameStream << std::setfill(L'0') << std::setw(framePadding) << frame;
+					result = std::regex_replace(result, std::wregex(frame_regex), frameStream.str().c_str());
 
-				// exporting animation as single file
-				if(isSequenceExportEnabled)
+					// Replace name after all operations, because it could match frame or extension regex
+					result = std::regex_replace(result, std::wregex(name_regex), fileName);
```

Comment:
If I'm not mistaken we woul have the same names for files with the same frame but different layer... Or did I missed something?